build/tasks/CreateCommonManifestTask.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Newtonsoft.Json.Linq;
+
+namespace RepoTasks
+{
+    public class CreateCommonManifestTask : Task
+    {
+        /// <summary>
+        /// The path for the common manifest file to be created.
+        /// </summary>
+        /// <returns></returns>
+        [Required]
+        public string DestinationFilePath { get; set; }
+
+        /// <summary>
+        /// The path for to the project.assets.json file for resolving package versions.
+        /// </summary>
+        /// <returns></returns>
+        [Required]
+        public string ProjectAssetsJsonFilePath { get; set; }
+
+        /// <summary>
+        /// The path for to the project.assets.json file for resolving package versions.
+        /// </summary>
+        /// <returns></returns>
+        [Required]
+        public ITaskItem[] Packages { get; set; }
+
+        public override bool Execute()
+        {
+            var projectAssets = JObject.Parse(File.ReadAllText(ProjectAssetsJsonFilePath));
+            var packages = projectAssets
+                    .SelectToken("$.targets.*") // target/.NETCoreApp,Version=v2.0
+                    .Children().Select(s => ((JProperty)s).Name); // Package.Name/Version
+
+            var xmlDoc = new XmlDocument();
+            var packagesElement = xmlDoc.CreateElement("StoreArtifacts");
+
+            foreach (var package in Packages.Select(i => i.ItemSpec))
+            {
+                var packageVersion = packages
```

Comment:
There may be cleaner ways of doing this. along with parsing out the Package.Name/Version from the project.assets.json lines 40-42